bundles/org.openhab.binding.irobot/src/main/java/org/openhab/binding/irobot/internal/handler/RoombaHandler.java

```diff
@@ -87,20 +110,59 @@
     private @Nullable Boolean vacHigh = null;
     private boolean isPaused = false;
 
+    private @Nullable Future<?> credentialRequester;
+    protected IRobotConnectionHandler connection = new IRobotConnectionHandler() {
+        @Override
+        public void receive(final String topic, final String json) {
+            RoombaHandler.this.receive(topic, json);
+        }
+
+        @Override
+        public void connectionStateChanged(MqttConnectionState state, @Nullable Throwable error) {
+            super.connectionStateChanged(state, error);
+            if (state == MqttConnectionState.CONNECTED) {
+                updateStatus(ThingStatus.ONLINE);
+            } else {
+                String message = (error != null) ? error.getMessage() : "Unknown reason";
+                updateStatus(OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, message);
+            }
+        }
+    };
+
     public RoombaHandler(Thing thing) {
         super(thing);
     }
 
     @Override
     public void initialize() {
-        config = getConfigAs(RoombaConfiguration.class);
-        updateStatus(ThingStatus.UNKNOWN);
-        scheduler.execute(this::connect);
+        IRobotConfiguration config = getConfigAs(IRobotConfiguration.class);
+
+        try {
+            InetAddress.getByName(config.getAddress());
```

Comment:
`initialize()` must return fast. This can block if the network fails. You could use `scheduler` to execute the lookup in a task.

Or can this be removed at all?

Reply:
removed